java/src/com/ibm/streamsx/topology/generator/spl/Preprocessor.java

```diff
@@ -110,6 +224,64 @@ private void relocateHashAdder(JsonObject hashAdder){
             // set Unparallel's output port schema using HashAdder's schema
             GraphUtilities.setOutputPortType(parent, 0, schema);
             GraphUtilities.setInputPortType(parent, 0, schema);
+        } else {
+            // Scenario #3: the parent $Unparallel$ has multiple children
+
+            // distinguish HashAdders from other children
+            LinkedList<JsonObject> HashAdders = new LinkedList<>();
+            LinkedList<JsonObject> others = new LinkedList<>();
+            children.forEach(
+                    (JsonObject parentChild) -> {
+                        if (GraphUtilities.isHashAdder(parentChild)) {
+                            HashAdders.add(parentChild);
+                        } else {
+                            others.add(parentChild);
+                        }
+                    }
+            );
+
+            // Please note that $Unparallel$ has at least one HashAdder child,
+            // because it is discovered through a HashAdder.
+            int nUnparallelCopy = 0;
+            for (JsonObject hashAdder: HashAdders) {
+                // 1. move the HashAdder upstream i.e.
+                // input -> $Unparallel$ -> HashAdder -> $Parallel$
+                // becomes
+                // input -> HashAdder -> $Parallel$
```

Comment:
Diagram at 250-251 is confusing/wrong? Isn't the `$Unparallel$` after the hash adder?

Reply:
Ok - I guess it wasn't clear this is a step by step explanation, not the final result, maybe an intro comment saying what the full transformation is and then how it is done step by step.